interview-questions.md

```diff
@@ -7,52 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: I like to think of props in React almost like variables specific to your components. Props are made in your "parent class" and can be passed down to children classes but not back up. You can create a prop with the syntax `propName = {this.whatEverYouArePassingHere}` inside a component call. We use it a lot to pass state or methods down. You can then call the props in whichever component you passed it to with `this.props.propNameHere`. 
 
-  Researched answer:
+  Researched answer: Props are read only, so if you pass a method as props you can not modify whatever the input is. When you modify the input directly it turns from a "pure" method/function to a "impure" method/function. You can avoid this with state and/or copies of your inputs.
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: A DOM event or just an event is used primarily when we make buttons to update something based on user interaction. Events are used a lot with props as you can activate methods passed as props within your component which can update your parent class. 
 
-  Researched answer:
+  Researched answer: DOM events perform certain actions depending on the users interaction on the page such as buttons. React events are camelCase and typed within curly braces. 
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: Object oriented programming is the main way we utilize React at LEARN. We create classes and methods instead of functions. It's not entirely different from functional programming, in functional programming you use functions instead of classes and objects. When functional programming, data isn't really stored inside objects and mainly transformed instead.
 
-  Researched answer:
+  Researched answer: Object oriented programming is the act of writing code with objects, classes, and methods. Functional programming utilizes functions instead, you can not store data in objects and have to transform it
```

Comment:
I'm gonna be honest and say you are going to want to take another look at this before taking these answers into an interview. 

OOP has four pillars. 

- Encapsulation
- Inheritance
- Polymorphism
- Abstraction

FP also has four pillars. How do they differ? What's the point?